### PR TITLE
Adds gameservers per node count and distribution

### PR DESCRIPTION
**Pull request by Kuqd**
Addresses #547 

I've included gameservers per nodes (count and distribution) in our `Agones GameServers` dashboard and move everything related to allocations to `Agones Allocations`.

### CODE REVIEW

File: pkg/metrics/controller_metrics.go [hunk 1]
```diff
@@ -111,6 +113,19 @@ var (
 			Aggregation: view.Count(),
 			TagKeys:     []tag.Key{keyType, keyFleetName},
 		},
+		&view.View{
```
Comment (jkowalski) [inline pkg/metrics/controller_metrics.go]: not sure if this is needed, the distribution bucket #0 will have the same value.

Comment (Kuqd) [reply pkg/metrics/controller_metrics.go]: Can I get the count of nodes currently in the cluster using the distribution ? I can get the average using _sum and _count but not sure about instant value.

Comment (jkowalski) [reply pkg/metrics/controller_metrics.go]: You can’t since the sum is ever increasing. A gauge is required.

Comment (Kuqd) [reply pkg/metrics/controller_metrics.go]: Ok so I’ll keep the view.

File: pkg/metrics/controller.go [hunk 2]
```diff
@@ -331,9 +343,44 @@ func (c *Controller) collectGameServerCounts() {
...
+		c.logger.WithError(err).Warn("failed listing gameservers")
+		return
+	}
+	recordWithTags(context.Background(), []tag.Mutator{tag.Insert(keyEmpty, "true")},
+		nodesCountStats.M(int64(len(nodes)-len(gsPerNodes))))
+	recordWithTags(context.Background(), []tag.Mutator{tag.Insert(keyEmpty, "false")},
+		nodesCountStats.M(int64(len(gsPerNodes))))
+
+	for _, node := range nodes {
+		if c, ok := gsPerNodes[node.Name]; ok {
```
[the remark this answers is not in the document]
Comment (Kuqd) [reply pkg/metrics/controller.go]: Interesting you found a bug here ! I was counting more than required, when I made the change metrics were different and correct.

**Pull request merged**